### PR TITLE
fix(sdk): fix issue of nested dict param defined

### LINKED ISSUE #490: ParallelLoop cannot handle list of dict defined as regular python variable.
/kind bug

**What steps did you take and what happened:**
Loop over a list of dictionaries works on containerOp output and pipeline parameters, but it doesn't work when the dictionary is defined in regular python variable. Take a look at the following example,
https://github.com/kubeflow/kfp-tekton/blob/master/sdk/python/tests/compiler/testdata/withitem_nested.py

Right now this example can work on list of str/int defined as regular python variable. However, the original example for `withitem_nested` is looping in a list of dict defined as regular python variable. We need to fix the compiler/controller to handle this edge case. 
```diff
diff --git a/sdk/python/tests/compiler/testdata/withitem_nested.py b/sdk/python/tests/compiler/testdata/withitem_nested.py
index 06935077..9db5b646 100644
--- a/sdk/python/tests/compiler/testdata/withitem_nested.py
+++ b/sdk/python/tests/compiler/testdata/withitem_nested.py
@@ -30,13 +30,13 @@ def get_code(self, ):
 
 @dsl.pipeline(name='my-pipeline')
 def pipeline(my_pipe_param: int = 10):
-    loop_args = [1, 2]
+    loop_args = [{'a': 1, 'b': 2}, {'a': 10, 'b': 20}]
     with dsl.ParallelFor(loop_args) as item:
         op1 = dsl.ContainerOp(
             name="my-in-coop1",
             image="library/bash:4.4.23",
             command=["sh", "-c"],
-            arguments=["echo op1 %s %s" % (item, my_pipe_param)],
+            arguments=["echo op1 %s %s" % (item.a, my_pipe_param)],
         )
 
         with dsl.ParallelFor([100, 200, 300]) as inner_item:
@@ -44,14 +44,14 @@ def pipeline(my_pipe_param: int = 10):
                 name="my-inner-inner-coop",
                 image="library/bash:4.4.23",
                 command=["sh", "-c"],
-                arguments=["echo op1 %s %s %s" % (item, inner_item, my_pipe_param)],
+                arguments=["echo op1 %s %s %s" % (item.a, inner_item, my_pipe_param)],
             )
 
         op2 = dsl.ContainerOp(
             name="my-in-coop2",
             image="library/bash:4.4.23",
             command=["sh", "-c"],
-            arguments=["echo op2 %s" % item],
+            arguments=["echo op2 %s" % item.b],
         )

```

**What did you expect to happen:**

**Additional information:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

* Python Version (use `python --version`):
* SDK Version: 
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by fenglixa**
**Which issue is resolved by this Pull Request:** 
Resolves #490

**Description of your changes:**
Handle the case of nested loop defined the dict parameters
**Environment tested:**
image

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [x] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

/cc @Tomcli

### CODE REVIEW

**Comment by Tomcli** [comment]
@fenglixa I'm not sure why travis ci is not running, can you push your changes in a new commit to run the travis test? thanks.

**Comment by Tomcli** [comment]
thanks @fenglixa 
/lgtm
/unhold
